device/sycl/src/clusterization/cluster_finding.cpp

```diff
@@ -0,0 +1,98 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// SYCL library include(s).
+#include "traccc/sycl/clusterization/cluster_finding.hpp"
+
+// SYCL library include(s).
+#include "component_connection.hpp"
+#include "measurement_creation.hpp"
+
+// Vecmem include(s).
+#include <vecmem/utils/sycl/copy.hpp>
+
+namespace traccc::sycl {
+
+cluster_finding::cluster_finding(vecmem::memory_resource &mr,
+                                 queue_wrapper queue)
+    : m_mr(mr), m_queue(queue) {}
+
+host_measurement_container cluster_finding::operator()(
+    const host_cell_container &cells_per_event) const {
+
+    // Number of modules
+    unsigned int num_modules = cells_per_event.size();
+
+    // Number of cells from all the modules
+    unsigned int cells_total_size = cells_per_event.total_size();
+
+    // Find the largest cell collection
+    // to estimate max. size of cluster (assuming that this cell collection
+    // would constitute to the whole cluster)
+    unsigned int max_cluster_size =
+        std::max_element(
+            cells_per_event.get_items().begin(),
+            cells_per_event.get_items().end(),
+            [](auto &c1, auto &c2) { return c1.size() < c2.size(); })
+            ->size();
+
+    // Cluster container buffer for the clusters and headers (cluster ids)
+    // assuming the worst case where 1 cell = 1 cluster
+    // and max. number of cells per module is max. number of cells per cluster
+    cluster_container_buffer clusters_buffer{
+        {cells_total_size, m_mr.get()},
+        {std::vector<std::size_t>(cells_total_size, 0),
+         std::vector<std::size_t>(cells_total_size, max_cluster_size),
+         m_mr.get()}};
+
+    // Vector for counts of clusters per each module
+    vecmem::vector<unsigned int> cluster_sizes(num_modules, 1, &m_mr.get());
+
+    // Atomic count of all clusters (needed inside component connection kernel
+    // but also here)
+    vecmem::vector<unsigned int> total_clusters(1, 0, &m_mr.get());
```

Comment:
So the `total_clusters` is needed anyway inside the component connection kernel (its intermediate atomic sums are used for correct indexing). 

For the `cluster_sizes`, using `copy.get_sizes()` doesn't really work here because what I need is the vector of "number of clusters per each module", where the get_sizes gives me numbers of cells in each cluster - it's the jagged vector of `vector<vector<cell>`.